dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/model/TaskNode.java

```diff
@@ -296,18 +304,49 @@ public void setTimeout(String timeout) {
     this.timeout = timeout;
   }
 
-  /**
-   * get task time out parameter
-   * @return task time out parameter
-   */
-  public TaskTimeoutParameter getTaskTimeoutParameter() {
-    if(StringUtils.isNotEmpty(this.getTimeout())){
-      String formatStr = String.format("%s,%s", TaskTimeoutStrategy.WARN.name(), TaskTimeoutStrategy.FAILED.name());
-      String taskTimeout = this.getTimeout().replace(formatStr,TaskTimeoutStrategy.WARNFAILED.name());
-      return JSONUtils.parseObject(taskTimeout,TaskTimeoutParameter.class);
+    public String getWaitStartTimeout() {
+        return waitStartTimeout;
+    }
+
+    public void setWaitStartTimeout(String waitStartTimeout) {
+        this.waitStartTimeout = waitStartTimeout;
+    }
+
+    /**
+     * get task time out parameter
+     *
+     * @return task time out parameter
+     */
+    public TaskTimeoutParameter getTaskTimeoutParameter() {
+        return parseTimeoutParameter(this.timeout);
+    }
+
+    /**
+     * get task timeout parameter for dependent node
+     *
+     * @return timeout parameter
+     */
+    public TaskTimeoutParameter getTaskTimeoutParameterForDependentNode() {
+        return parseTimeoutParameter(this.waitStartTimeout);
+    }
+
+    /**
+     * parse string to timeout parameter instance
+     * @param timeoutParameterString timeout description string
+     * @return TaskTimeoutParameter
+     */
+    private TaskTimeoutParameter parseTimeoutParameter(String timeoutParameterString) {
+        TaskTimeoutParameter taskTimeoutParameter = null;
```

Comment:
Will it better to make the `parseTimeoutParameter` static method?
Because this method make a new instance inner.

Reply:
Yes, of course.